common/ast/tests/it/testdata/statement.txt

```diff
@@ -3352,3 +3352,430 @@ CreateTable(
+            level: Database(
+                None,
+            ),
+        },
+        principal: User(
+            UserIdentity {
+                username: "test-grant",
+                hostname: "localhost",
+            },
+        ),
+    },
+)
+
+
+---------- Input ----------
+GRANT SELECT, CREATE ON * TO ROLE 'role1';
+---------- Output ---------
+GRANT SELECT, CREATE ON TO ROLE role1
+---------- AST ------------
+Grant(
+    GrantStatement {
+        source: Privs {
+            privileges: [
+                Select,
+                Create,
+            ],
+            level: Database(
+                None,
+            ),
+        },
+        principal: Role(
+            "role1",
+        ),
+    },
+)
+
+
+---------- Input ----------
+GRANT ALL ON *.* TO 'test-grant'@'localhost';
+---------- Output ---------
+GRANT ALL PRIVILEGES ON *.* TO USER 'test-grant'@'localhost'
+---------- AST ------------
+Grant(
+    GrantStatement {
+        source: ALL {
+            level: Global,
+        },
+        principal: User(
+            UserIdentity {
+                username: "test-grant",
+                hostname: "localhost",
+            },
+        ),
+    },
+)
+
+
+---------- Input ----------
+GRANT ALL ON *.* TO ROLE 'role2';
+---------- Output ---------
+GRANT ALL PRIVILEGES ON *.* TO ROLE role2
+---------- AST ------------
+Grant(
+    GrantStatement {
+        source: ALL {
+            level: Global,
+        },
+        principal: Role(
+            "role2",
+        ),
+    },
+)
+
+
+---------- Input ----------
+GRANT ALL PRIVILEGES ON * TO 'test-grant'@'localhost';
+---------- Output ---------
+GRANT ALL PRIVILEGES ON TO USER 'test-grant'@'localhost'
+---------- AST ------------
+Grant(
+    GrantStatement {
+        source: ALL {
+            level: Database(
+                None,
+            ),
+        },
+        principal: User(
+            UserIdentity {
+                username: "test-grant",
+                hostname: "localhost",
+            },
+        ),
+    },
+)
+
+
+---------- Input ----------
+GRANT ALL PRIVILEGES ON * TO ROLE 'role3';
+---------- Output ---------
+GRANT ALL PRIVILEGES ON TO ROLE role3
+---------- AST ------------
+Grant(
+    GrantStatement {
+        source: ALL {
+            level: Database(
+                None,
+            ),
+        },
+        principal: Role(
+            "role3",
+        ),
+    },
+)
+
+
+---------- Input ----------
+GRANT ROLE 'test' TO 'test-user';
+---------- Output ---------
+GRANT ROLE test TO USER 'test-user'@'%'
+---------- AST ------------
+Grant(
+    GrantStatement {
+        source: Role {
+            role: "test",
+        },
+        principal: User(
+            UserIdentity {
+                username: "test-user",
+                hostname: "%",
+            },
+        ),
+    },
+)
+
+
+---------- Input ----------
+GRANT ROLE 'test' TO USER 'test-user';
+---------- Output ---------
+GRANT ROLE test TO USER 'test-user'@'%'
```

Comment:
The output is mismatched.

Reply:
I think that's ok because the user not specified host will set to default value %